dotnet/src/Connectors/Connectors.Memory.Postgres/README.md

```diff
@@ -16,15 +20,84 @@ How to install the pgvector extension, please refer to its [documentation](https
 docker run -d --name postgres-pgvector -p 5432:5432 -e POSTGRES_PASSWORD=mysecretpassword ankane/pgvector
 ```
 
-2. To use Postgres as a semantic memory store:
+2. Create database and enable pgvector extension on database
+
+```bash
+docker exec -it pgvector psql -U postgres
+
+postgres=# CREATE DATABASE sk_demo;
+postgres=# \c sk_demo
+sk_demo=# CREATE EXTENSION vector;
+```
+
+> Note, "Azure Cosmos DB for PostgreSQL" uses `SELECT CREATE_EXTENSION('vector');` to enable the extension.
+
+3. To use Postgres as a semantic memory store:
 
 ```csharp
-using PostgresMemoryStore memoryStore = await PostgresMemoryStore.ConnectAsync("Host=localhost;Port=5432;Database=sk_memory;User Id=postgres;Password=mysecretpassword", vectorSize: 1536);
+NpgsqlDataSourceBuilder dataSourceBuilder = new NpgsqlDataSourceBuilder("Host=localhost;Port=5432;Database=sk_memory;User Id=postgres;Password=mysecretpassword");
+dataSourceBuilder.UseVector();
+NpgsqlDataSource dataSource = dataSourceBuilder.Build();
+
+PostgresMemoryStore memoryStore = new PostgresMemoryStore(dataSource, vectorSize: 1536/*, schema: "public" */);
 
 IKernel kernel = Kernel.Builder
     .WithLogger(ConsoleLogger.Log)
-    .Configure(c => c.AddOpenAITextEmbeddingGenerationService("text-embedding-ada-002", Env.Var("OPENAI_API_KEY")))
+    .WithOpenAITextEmbeddingGenerationService("text-embedding-ada-002", Env.Var("OPENAI_API_KEY"))
     .WithMemoryStorage(memoryStore)
+    //.WithPostgresMemoryStore(dataSource, vectorSize: 1536) // This method offers an alternative approach to registering Postgres memory store.
     .Build();
 ```
 
+## Migration from older versions
+Since Postgres Memory connector has been re-implemented, the new implementation uses a separate table to store each Collection.
+
+We provide the following migration script to help you migrate to the new structure. However, please note that due to the use of collections as table names, you need to make sure that all Collections conform to the [Postgres naming convention](https://www.postgresql.org/docs/15/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS) before migrating.
+
+- Table names may only consist of ASCII letters, digits, and underscores.
+- Table names must start with a letter or an underscore.
+- Table names may not exceed 63 characters in length.
+- Table names are case-insensitive, but it is recommended to use lowercase letters.
+
+```sql
+-- Create new tables, each with the name of the collection field value
+DO $$
+DECLARE
+    r record;
+BEGIN
+    FOR r IN SELECT DISTINCT collection FROM sk_memory_table LOOP
+
+        -- Drop Table
+        EXECUTE format('DROP TABLE IF EXISTS %I;', r.collection);
```

Comment:
Thank you for this migration script!
One note: we should not drop user existing tables. If previously user had `contacts` collection in `sk_memory_table` and separate `contacts` table in Postgres database - during migration script execution we should throw an error that table with the same name already exists, so user will need to decide what to do next, leave it as it is or drop table manually. 
Migration script should not brake user's current DB architecture.

Reply:
I commented out this sql block, and added a comment, just like drop sk_memory_table, to let the user decide whether to enable it or not, so that the script can be executed multiple times. What do you think?

Reply:
> I commented out this sql block, and added a comment, just like drop sk_memory_table, to let the user decide whether to enable it or not, so that the script can be executed multiple times. What do you think?

Sounds awesome, thank you!